### ISSUE TITLE
Scalatra 2.4 TODOs

### ISSUE THREAD

**Issue by seratch** [scalatra/scalatra#489]
I'll work on RCs release for 2.4.0 soon. The following things should be done too. Please let me know if you find other tasks I missed.

#### 2.4 TODOs

- [x] scalate 1.7.1 https://github.com/scalatra/scalatra/pull/476
- [x] scalatra-sbt 0.4? built with xsbt-web-plugin 1.0.0 https://github.com/scalatra/scalatra-sbt/issues/9 https://github.com/scalatra/scalatra-sbt/pull/12
- [x] json4s 3.3.0+ (I'll work on this too) https://github.com/json4s/json4s/pull/224 -> 3.3.0
- [x] atmosphere 2.2 upgrade (not only for 2.3.x) https://github.com/scalatra/scalatra/pull/487
- [x] NPE on JacksonJsonSupport since 2.4.0.M3 #486
- [x] Params and multiParams are empty for PATCH request #531
- [x] scalatra-json issue #548
- [x] merge https://github.com/scalatra/scalatra-sbt.g8/pull/49

#### Moved to 2.5

- Swagger 2.0 support #495

**Comment by tminglei** [comment]
Any plan for swagger?

**Comment by seratch** [comment]
@tminglei Do you mean swagger spec 2.0 support? 

To be honest, I'm not a heavy user of scalatra-swagger. If someone would like to work on that, we would be very grateful for any help.

**Comment by tminglei** [comment]
Yes, swagger spec 2.0 support.
_Scalatra swagger is the best elegant implementation I saw._

**Comment by rossabaker** [comment]
We've wanted to use some 2.0 features at work.  I would love to see this, but I have not sized the work to see how difficult it would be.

Otherwise, :+1: to @seratch's list.

**Comment by seratch** [comment]
TODO: json4s 3.3.0.RC1 breaks scalatra-swagger https://github.com/scalatra/scalatra/pull/490

**Comment by mdedetrich** [comment]
Is json4s 3.3.0 something that is set in stone? Only asking this because we have a lot of libraries that still use 3.1.0, and this will break binary compatibility

We are currently using 2.4.0-M2, which is using an older Json4s version

**Comment by seratch** [comment]
We understand json4s's binary compatibility issue well... So now json4s team is planning to use mima since 3.3.0 release. https://github.com/json4s/json4s/issues/225

Scalatra 2.3.x already depends on json4s 3.2.11 which is binary incompatible even with 3.2.10. I'm afraid your llibs that depends on json4s 3.1.0 may not work with Scalatra 3.2.x.

Even after Scalatra 2.4.0 release, I think it would be needed to keep maintaining Scalatra 2.3.x series for a while. The relationship between Scalatra and json4s will become apparent below.

- Scalatra 2.4 is compatible with json4s 3.3
- Scalatra 2.3 is compatible with json4s 3.2

**Comment by mdedetrich** [comment]
Ok cool, hopefully the libraries we use will migrate to latest JSON4s and good to here that the json4s team will start using mima

**Comment by LKay** [comment]
Any plans on making converting json response into more abstract layer? Now it is heavily bounded with json4s but it would be nice to have a choice to use spray json, play json etc. instead.

**Comment by dozed** [comment]
Generally I like to have support for other json libraries in scalatra-json. That would be something for 2.5.x?

**Comment by seratch** [comment]
All the existing issues have been fixed but #548 should be fixed before 2.4.0 release although I haven't checked it out in detail yet.

https://groups.google.com/forum/#!topic/scalatra-user/V0dS46b-jfU

**Comment by seratch** [comment]
#548 has been closed. Is it about time to release 2.4.0?

**Comment by seratch** [comment]
https://gitter.im/scalatra/scalatra?at=56563c0a63bfb30b58e47445

```
seratch Nov 26 07:54
What do you think? https://github.com/scalatra/scalatra/issues/489#issuecomment-159749727

rossabaker Nov 26 07:55
I think we're in good shape.

dozed Nov 26 08:01
Looks good. We should add that to the release notes. Also some words about scalatra-sbt using xsbt-web-plugin 2 would be helpful.
```

@rossabaker @dozed Sorry for delay. I'm going to work on 2.4.0 release this week.

**Comment by dozed** [comment]
I added this PR to the TODO list https://github.com/scalatra/scalatra-sbt.g8/pull/49.

**Comment by seratch** [comment]
2.4.0 is out. scalatra/scalatra-sbt.g8#49 has been merged right now (sorry for that). Let me close this issue. Thank you all.
